quinn/src/send_stream.rs

```diff
@@ -307,6 +307,19 @@ impl Future for Stopped<'_> {
     }
 }
 
+impl<'a> Drop for Stopped<'a> {
+    fn drop(&mut self) {
+        let mut conn = self.stream.conn.state.lock("Stopped::drop");
+
+        if let Some(stopped) = conn.stopped.remove(&self.stream.stream) {
+            // Given this is the future that would have registered the waker we may not need to
+            // actually call Waker::wake, calling here may result in a spurious wake up but at
+            // least we can be certain that there isn't a task that will never be woken up.
+            stopped.wake();
+        }
```

Comment:
Is there a specific case of a never-woken task that you envision? I'm having trouble coming up with any cases where calling wake here is necessary.

Reply:
I don't have a specific case, and would be completely comfortable just ignoring the removed waker. Just let me know what you'd prefer we do here and i'll make the change.

Reply:
In the interest of maintainability, let's ignore it then.